tests/pkg_tensorflow-lite/mnist/main_functions.cc

```diff
@@ -0,0 +1,127 @@
+    tflite::MicroInterpreter* interpreter = nullptr;
+    TfLiteTensor* input = nullptr;
+    TfLiteTensor* output = nullptr;
+
+    // Create an area of memory to use for input, output, and intermediate arrays.
+    // Finding the minimum value for your model may require some trial and error.
+    constexpr int kTensorArenaSize = 6 * 1024;
+    uint8_t tensor_arena[kTensorArenaSize];
+}  // namespace
+
+// The name of this function is important for Arduino compatibility.
+void setup()
+{
+    // Set up logging. Google style is to avoid globals or statics because of
+    // lifetime uncertainty, but since this has a trivial destructor it's okay.
+    static tflite::MicroErrorReporter micro_error_reporter;
+    error_reporter = &micro_error_reporter;
+
+    // Map the model into a usable data structure. This doesn't involve any
+    // copying or parsing, it's a very lightweight operation.
+    model = tflite::GetModel(model_tflite);
+    if (model->version() != TFLITE_SCHEMA_VERSION) {
+        printf("Model provided is schema version %d not equal "
+               "to supported version %d.",
+               static_cast<uint8_t>(model->version()), TFLITE_SCHEMA_VERSION);
+        return;
+    }
+
+    // Explicitly load required operators
+    static tflite::MicroMutableOpResolver micro_mutable_op_resolver;
+    micro_mutable_op_resolver.AddBuiltin(
+        tflite::BuiltinOperator_FULLY_CONNECTED,
+        tflite::ops::micro::Register_FULLY_CONNECTED(), 1, 4);
+    micro_mutable_op_resolver.AddBuiltin(
+        tflite::BuiltinOperator_SOFTMAX,
+        tflite::ops::micro::Register_SOFTMAX(), 1, 2);
+    micro_mutable_op_resolver.AddBuiltin(
+        tflite::BuiltinOperator_QUANTIZE,
+        tflite::ops::micro::Register_QUANTIZE());
+    micro_mutable_op_resolver.AddBuiltin(
+        tflite::BuiltinOperator_DEQUANTIZE,
+        tflite::ops::micro::Register_DEQUANTIZE(), 1, 2);
+
+    // Build an interpreter to run the model with.
+    static tflite::MicroInterpreter static_interpreter(
+        model, micro_mutable_op_resolver, tensor_arena, kTensorArenaSize, error_reporter);
+    interpreter = &static_interpreter;
+
+    // Allocate memory from the tensor_arena for the model's tensors.
+    TfLiteStatus allocate_status = interpreter->AllocateTensors();
+    if (allocate_status != kTfLiteOk) {
+        puts("AllocateTensors() failed");
+        return;
+    }
+
+    // Obtain pointers to the model's input and output tensors.
+    input = interpreter->input(0);
+    output = interpreter->output(0);
+
+    // Copy digit array in input tensor
+    for (unsigned i = 0; i < digit_len; ++i) {
+        input->data.f[i] = static_cast<float>(digit[i]) / 255.0;
+    }
+
+    // Run inference, and report any error
+    TfLiteStatus invoke_status = interpreter->Invoke();
+    if (invoke_status != kTfLiteOk) {
+        puts("Invoke failed");
+        return;
+    }
+
+    // Get the best match from the output tensor
+    float val = 0;
```

Comment:
I assume this needs floats?

Reply:
Yes, float are used a lot during inference. Why are you asking ?